src/Storage/DbalNestedSet.php

```diff
@@ -349,17 +355,37 @@ public function moveSubTreeAfter(Node $target, Node $node) {
    *   If a transaction error occurs.
    */
   protected function moveSubTreeToPosition($newLeftPosition, Node $node, $newDepth) {
+    $this->moveMultipleSubTreesToPosition($newLeftPosition, [$node], $newDepth);
+  }
+
+  /**
+   * Moves multiple subtrees to a new position.
+   *
+   * @param int $newLeftPosition
+   *   The new left position.
+   * @param \PNX\NestedSet\Node[] $nodes
+   *   The node to move.
+   * @param int $newDepth
+   *   Depth of new position.
+   *
+   * @throws \Exception
+   *   If a transaction error occurs.
+   */
+  protected function moveMultipleSubTreesToPosition($newLeftPosition, array $nodes, $newDepth) {
     try {
+
+      $firstNode = $nodes[0];
+      $lastNode = $nodes[count($nodes) - 1];
       // Calculate position adjustment variables.
-      $width = $node->getRight() - $node->getLeft() + 1;
-      $distance = $newLeftPosition - $node->getLeft();
-      $tempPos = $node->getLeft();
+      $width = $lastNode->getRight() - $firstNode->getLeft() + 1;
```

Comment:
We only call this internally - but should we be checking that $nodes are consecutive and ordered?